pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -2,31 +2,33 @@ package kvstore
 
 import (
 	"context"
-	"errors"
-	"fmt"
 	"time"
 
 	"github.com/grafana/grafana/pkg/infra/kvstore"
 	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	"github.com/grafana/grafana/pkg/services/secrets"
 	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
 )
 
 const (
 	// Wildcard to query all organizations
 	AllOrganizations = -1
 )
 
+var logger = log.New("secrets.kvstore")
```

Comment:
While I am not personally bothered by using a global logger, I have seen PR feedback that seems to be against doing this. Should we just pass the logger into the helper functions instead?

Reply:
Since there are a lot of functions that are not part of an interface that use `logger` I was thinking it would be best to keep it as a global instead of passing it as an argument to keep the code cleaner.

Reply:
Done on 1ccdedf433